### PR TITLE
Allow prometheus to gather metrics published by controller-runtime

### LINKED ISSUE #53: Add metrics to reflect controller health
Possible metrics:

- Reconciliation latency
- Reconciliation failures 
- JobSets waiting to be processed (we need to look at controller-runtime if it exposes something)

### PR DESCRIPTION
**Pull request by danielvegamyhre**
Fixes #53 

Example querying `controller_runtime_reconcile_errors_total` metric, and jobset controller is in the results:

```
$ curl 'http://localhost:9090/api/v1/query?query=controller_runtime_reconcile_errors_total'

{"status":"success","data":{"resultType":"vector","result":[{"metric":{"__name__":"controller_runtime_reconcile_errors_total","container":"kube-rbac-proxy","controller":"cert-rotator","endpoint":"https","instance":"10.68.0.118:8443","job":"jobset-controller-manager-metrics-service","namespace":"jobset-system","pod":"jobset-controller-manager-75c4c8fc5f-tc4wz","service":"jobset-controller-manager-metrics-service"},"value":[1682970346.084,"0"]},{"metric":{"__name__":"controller_runtime_reconcile_errors_total","container":"kube-rbac-proxy","controller":"jobset","endpoint":"https","instance":"10.68.0.118:8443","job":"jobset-controller-manager-metrics-service","namespace":"jobset-system","pod":"jobset-controller-manager-75c4c8fc5f-tc4wz","service":"jobset-controller-manager-metrics-service"},"value":[1682970346.084,"0"]}]}}
```

### CODE REVIEW

**Comment by ahg-g** [comment]
Do we need to change the makefile to apply the prometheus manifests?

See https://kueue.sigs.k8s.io/docs/installation/#add-metrics-scraping-for-prometheus-operator-1

**Comment by danielvegamyhre** [comment]
@ahg-g Currently the prometheus manifests are applied as part of the default installation of `config/default` due to this line [here](https://github.com/kubernetes-sigs/jobset/pull/99/files#diff-c006b1ef995eeeb57408cce42bd84db8dfc0cbe18df03222857665cebb1ac6fdR25). However, I can change this to follow Kueue's example of only applying prometheus manifests when the user explicitly intends to (i.e. by running a make command), what do you think?

File: config/prometheus/monitor.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ metadata:
     app.kubernetes.io/part-of: jobset
     app.kubernetes.io/managed-by: kustomize
   name: controller-manager-metrics-monitor
-  namespace: system
```
Comment (ahg-g) [inline config/prometheus/monitor.yaml]: shouldn't this be overridden by kustomize?

File: config/components/manager/kustomization.yaml [hunk 2]
```diff
@@ -5,5 +5,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: gcr.io/k8s-staging-jobset/jobset
+  newName: gcr.io/danielvm-gke-dev2/jobset
```
Comment (ahg-g) [inline config/components/manager/kustomization.yaml:8]: revert :)

File: config/prometheus/monitor.yaml (hunk 1 above)
Comment (danielvegamyhre) [reply config/prometheus/monitor.yaml]: yes, after the reorganization of the config directory i just pushed, the override occurs in `config/prometheus/kustomization.yaml`. Users can apply prometheus manifests individually with `make prometheus`

**Comment by danielvegamyhre** [comment]
@ahg-g I reorganized the `config` directory to allow prometheus manifests can be applied separately via `make prometheus`, rather than included in the default installation. This is ready for another look.

File: docs/setup/install.md [hunk 3]
```diff
@@ -132,15 +132,14 @@ cd jobset
...
-<!-- to scrape metrics from jobset components, run the following command: -->
+To allow [prometheus-operator](https://github.com/prometheus-operator/prometheus-operator)
+to scrape metrics from jobset components, run the following command:
 
-<!-- ```shell -->
-<!-- make prometheus -->
-<!-- ``` -->
+```shell
+make prometheus
+```
```
Comment (ahg-g) [inline docs/setup/install.md:142]: nothing applicable to the development version?

Comment (danielvegamyhre) [reply docs/setup/install.md:142]: can you clarify what you mean?

Comment (ahg-g) [reply docs/setup/install.md:142]: we have instructions to deploy jobset from the latest pre-built development version above: "Install the latest development version"; how would one install prometheus if they want to go that route?

Comment (danielvegamyhre) [reply docs/setup/install.md:151]: I see. Updated this section accordingly.

**Pull request merged**